### ISSUE TITLE
what did you intend to do with drug_name.split("+")]

### ISSUE THREAD

**Issue by bhomass** [theislab/chemCPA#132]
in data.py, there is this statement

```
for d in self.drugs_names:
    [drugs_names_unique.add(i) for i in d.split("+")]
```

It led to the code bombing in the following line
```
    name_to_smiles_map = {
        drug: canonicalize_smiles(smiles)
        for drug, smiles in dataset.obs.groupby(
            [perturbation_key, smiles_key]
        ).groups.keys()
    }
```
Upon examination of the drug names, there are only 4 drug names that contains '+'
(+)-3-(1-propyl-piperidin-3-yl)-phenol
(+|-)-7-hydroxy-2-(N,N-di-n-propylamino)tetralin
flurbiprofen-(+|-)
atenolol-(+|-)

I assume the sensible thing to do would be to eliminate (+) or (+|-) and the trailing or preceding -. 
But  [drugs_names_unique.add(i) for i in d.split("+")] would not be doing that. It would simply leave fragments like '(' as a possible drug name. 

If someone can point out if my interpretation is correct.

The comment for drug_names_to_once_canon_smiles() says
#This function will need to be rewritten to handle datasets with combinations
but I don't get what is meant by "combinations". Are there some drugs that uses '+' to combine multiple formula together, and that is why you are doing split('+'). If so, the (+) cases should be exemplified from the split processing. But I don't see that mechanism in place.

**Comment by bhomass** [comment]
I used the re split with 
`plus_pattern = r'(?<!\()\+'`

and it worked.

**Comment by MxMstrmn** [comment]
The "+" sign is meant for indicating drug combinations, this can be done with chemCPA as demonstrated [here](https://www.embopress.org/doi/full/10.15252/msb.202211517). I will make this more clear in a future PR

**Comment by bhomass** [comment]
yes, I do understand the intent for the '+' in the code. my point is the code for splitting by '+' will fail for the dataset, because of drug names that already contains '+', as in (+)-3-(1-propyl-piperidin-3-yl)-phenol

**Comment by sepidism** [comment]
Were you able to run manual_seml_sweep.py? I keep getting random errors regarding the data. I'm trying with sciplex_complete_middle_subset.h5ad and slincs_full_smiles_sciplex_genes.h5ad.

**Comment by bhomass** [comment]
This is not a repo you can just download and run. There are  a hundred and
one little bugs mismatches discrepancies throughout. But I worked through
the errors one by one and got it running at the end.

**Comment by bhomass** [comment]
apparently the drug names with the funny + were eliminated during preprocessing, if you were able to run through the code in the preprocessing folder. For us, that is not possible due to unposted input files.

**Comment by MxMstrmn** [comment]
Hi @bhomass, 

why were you not able to remove those '+' signs from the drug names? What do you mean by `unposted input files`?

**Comment by bhomass** [comment]
Yes, I did remove the drugs with + in the names once I realized that is how
you handled these.

I posted a few times all the missing input files that show up in the code
but are not in the download links.

Let me walk through all the notebook code in the preprocessing folder one
more time and make a list of all input files that are missing,
